### PR TITLE
Add documentation, switch to `pyproject.toml` and `ruff`

### PR DESCRIPTION
**Pull request by araffin**
## Description

Add proper documentation because the README was a mess.

Also do the same as in https://github.com/DLR-RM/stable-baselines3/pull/1361

Can be see here: https://rl-baselines3-zoo.readthedocs.io/en/feat-doc/index.html

@qgallouedec in case you can do a quick review that's nice, otherwise I'll merge it without as this is important to provide clearer doc.

You don't need to go though all files, there are too many changes.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

See #299 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+# a list of builtin themes.
+
+# Fix for read the docs
+on_rtd = os.environ.get("READTHEDOCS") == "True"
+if on_rtd:
+    html_theme = "default"
+else:
+    html_theme = "sphinx_rtd_theme"
+
+html_logo = "../images/car.jpg"
```
Comment (qgallouedec) [inline docs/conf.py:138]: why do you add `logo.png` if you use `car.jpg` here?

**Comment by qgallouedec** [comment]
Looks great! It's much better now!
I didn't go through all the files since it's unlikely to break anything.

**Review by qgallouedec — APPROVED** [review]

File: docs/conf.py (hunk 1 above)
Comment (araffin) [reply docs/conf.py:138]: logo is a remain from SB3 doc... will delete

**Comment by araffin** [comment]
>  I didn't go through all the files since it's unlikely to break anything.

The build is somehow broken now, but related to setuptools I think.

**Comment by qgallouedec** [comment]
It seems like enjoy is broken for PandaReach. I'm taking a look

**Comment by araffin** [comment]
> It seems like enjoy is broken for PandaReach. I'm taking a look

The problem comes from `VecNormalize` which overrides somehow the observation space. Will see if I can push a fix soon.

**Comment by araffin** [comment]
I found the issue, it is a mix of https://github.com/DLR-RM/rl-trained-agents/pull/10 and https://github.com/DLR-RM/stable-baselines3/commit/12e9917c24dc23d7de7694a924f017c6a8e9a6ce (where the observation space was set to the env one), I will push a fix for the VecNormalize stats.

**Pull request merged**